src/main/java/org/opensearch/knn/index/query/nativelib/NativeEngineKnnVectorQuery.java

```diff
@@ -63,7 +63,9 @@ public Weight createWeight(IndexSearcher indexSearcher, ScoreMode scoreMode, flo
         } else {
             int firstPassK = rescoreContext.getFirstPassK(finalK);
             perLeafResults = doSearch(indexSearcher, leafReaderContexts, knnWeight, firstPassK);
-            ResultUtil.reduceToTopK(perLeafResults, firstPassK);
+            if (KNNSettings.isShardLevelRescoringDisabledForDiskBasedVector(knnQuery.getIndexName()) == false) {
+                ResultUtil.reduceToTopK(perLeafResults, firstPassK);
+            }
```

Comment:
can we have some logs may be deubg logs/warm logs to say that shard level rescoring is disabled.

Reply:
Hahaha , I just deleted from Jack comment .....